concrete-core/src/backends/core/private/crypto/glwe/keyswitch.rs

```diff
@@ -0,0 +1,887 @@
+#[cfg(feature = "serde_serialize")]
+use serde::{Deserialize, Serialize};
+
+use concrete_commons::dispersion::DispersionParameter;
+use concrete_commons::key_kinds::BinaryKeyKind;
+use concrete_commons::numeric::SignedInteger;
+use concrete_commons::parameters::{
+    CiphertextCount, DecompositionBaseLog, DecompositionLevelCount, GlweDimension, GlweSize,
+    LweDimension, MonomialDegree, PlaintextCount, PolynomialSize,
+};
+
+use crate::backends::core::private::crypto::encoding::PlaintextList;
+use crate::backends::core::private::crypto::lwe::{LweCiphertext, LweList};
+use crate::backends::core::private::crypto::secret::generators::EncryptionRandomGenerator;
+use crate::backends::core::private::crypto::secret::{GlweSecretKey, LweSecretKey};
+use crate::backends::core::private::math::decomposition::{
+    DecompositionLevel, DecompositionTerm, SignedDecomposer,
+};
+use crate::backends::core::private::math::tensor::{
+    ck_dim_div, ck_dim_eq, tensor_traits, AsMutSlice, AsMutTensor, AsRefSlice, AsRefTensor, Tensor,
+};
+use crate::backends::core::private::math::torus::UnsignedTorus;
+
+use super::{GlweCiphertext, GlweList};
+
+/// A packing keyswithing key.
+///
+/// A packing keyswitching key allows to  pack several LweCiphertext into a single GlweCiphertext.
+#[cfg_attr(feature = "serde_serialize", derive(Serialize, Deserialize))]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PackingKeyswitchKey<Cont> {
+    tensor: Tensor<Cont>,
+    decomp_base_log: DecompositionBaseLog,
+    decomp_level_count: DecompositionLevelCount,
+    glwe_size: GlweSize,
```

Comment:
As opposed to what? There's only one glwe size involved. I think using "output" might be confusing, since the user could expect to find an input_glwe_size.